src/Srclib/Converter.hs

```diff
@@ -193,3 +197,17 @@ fetcherToDepType fetcher | depTypeToFetcher URLType == fetcher = Just URLType
 fetcherToDepType fetcher | depTypeToFetcher UserType == fetcher = Just UserType
 fetcherToDepType fetcher | depTypeToFetcher PubType == fetcher = Just PubType
 fetcherToDepType _ = Nothing
+
+-- | Merge two license source units, keeping the name and type from the first unit
+mergeLicenseSourceUnits :: Maybe LicenseSourceUnit -> Maybe LicenseSourceUnit -> Maybe LicenseSourceUnit
+mergeLicenseSourceUnits maybeFirst maybeSecond =
+  case (maybeFirst, maybeSecond) of
+    (Nothing, Nothing) -> Nothing
+    (Just first, Nothing) -> Just first
+    (Nothing, Just second) -> Just second
+    (Just first, Just second) ->
+      Just first{licenseSourceUnitLicenseUnits = mergedLicenseUnits}
+      where
+        firstLicenseUnits = licenseSourceUnitLicenseUnits first
+        secondLicenseUnits = licenseSourceUnitLicenseUnits second
+        mergedLicenseUnits = foldr NE.cons firstLicenseUnits secondLicenseUnits
```

Comment:
In general functional-style programs prefer concrete operations on concrete sets of data, and leave monads like `Maybe` up to composition. Prefer to leave it up to your user to compose these, e.g. via `fromMaybe`, their own pattern matching, or something else. This makes your function more composable and simpler.

A similar concept in other languages is Rust, which uses its `Option` "monad" similarly. You don't generally create a function signature like `Fn(Option<A>) -> Option<B>`; instead you typically create a function signature like `Fn(A) -> B` and then use `Option::map` to compose it. Same idea here, it's just here it's `fmap` and it's generic over all functors.